diff -Naur a/drivers/base/regmap/regmap.c b/drivers/base/regmap/regmap.c
--- a/drivers/base/regmap/regmap.c	2023-09-16 20:43:35.384016595 +0200
+++ b/drivers/base/regmap/regmap.c	2023-09-16 20:43:35.388016649 +0200
@@ -263,6 +263,30 @@
 	b[0] = cpu_to_le32(val << shift);
 }
 
+static void regmap_format_32_lbe(void *buf, unsigned int val, unsigned int shift)
+{
+	u8 *b = buf;
+
+	val <<= shift;
+
+	b[0] = val >> 16;
+	b[1] = val >> 24;
+	b[2] = val;
+	b[3] = val >> 8;
+}
+
+static void regmap_format_32_ble(void *buf, unsigned int val, unsigned int shift)
+{
+	u8 *b = buf;
+
+	val <<= shift;
+
+	b[0] = val >> 8;
+	b[1] = val;
+	b[2] = val >> 24;
+	b[3] = val >> 16;
+}
+
 static void regmap_format_32_native(void *buf, unsigned int val,
 				    unsigned int shift)
 {
@@ -359,6 +383,28 @@
 	return le32_to_cpu(b[0]);
 }
 
+static unsigned int regmap_parse_32_lbe(const void *buf)
+{
+	const u8 *b = buf;
+	unsigned int ret = b[2];
+	ret |= ((unsigned int)b[3]) << 8;
+	ret |= ((unsigned int)b[0]) << 16;
+	ret |= ((unsigned int)b[1]) << 24;
+
+	return ret;
+}
+
+static unsigned int regmap_parse_32_ble(const void *buf)
+{
+	const u8 *b = buf;
+	unsigned int ret = b[1];
+	ret |= ((unsigned int)b[0]) << 8;
+	ret |= ((unsigned int)b[3]) << 16;
+	ret |= ((unsigned int)b[2]) << 24;
+
+	return ret;
+}
+
 static void regmap_parse_32_be_inplace(void *buf)
 {
 	__be32 *b = buf;
@@ -373,6 +419,20 @@
 	b[0] = le32_to_cpu(b[0]);
 }
 
+static void regmap_parse_32_lbe_inplace(void *buf)
+{
+	u32 *b = buf;
+
+	b[0] = regmap_parse_32_lbe(buf);
+}
+
+static void regmap_parse_32_ble_inplace(void *buf)
+{
+	u32 *b = buf;
+
+	b[0] = regmap_parse_32_ble(buf);
+}
+
 static unsigned int regmap_parse_32_native(const void *buf)
 {
 	return *(u32 *)buf;
@@ -875,6 +935,16 @@
 			map->format.parse_val = regmap_parse_32_le;
 			map->format.parse_inplace = regmap_parse_32_le_inplace;
 			break;
+		case REGMAP_ENDIAN_LITTLE_BIG:
+			map->format.format_val = regmap_format_32_lbe;
+			map->format.parse_val = regmap_parse_32_lbe;
+			map->format.parse_inplace = regmap_parse_32_lbe_inplace;
+			break;
+		case REGMAP_ENDIAN_BIG_LITTLE:
+			map->format.format_val = regmap_format_32_ble;
+			map->format.parse_val = regmap_parse_32_ble;
+			map->format.parse_inplace = regmap_parse_32_ble_inplace;
+			break;
 		case REGMAP_ENDIAN_NATIVE:
 			map->format.format_val = regmap_format_32_native;
 			map->format.parse_val = regmap_parse_32_native;
diff -Naur a/include/linux/regmap.h b/include/linux/regmap.h
--- a/include/linux/regmap.h	2023-09-16 20:43:35.380016541 +0200
+++ b/include/linux/regmap.h	2023-09-16 20:43:35.388016649 +0200
@@ -143,6 +143,10 @@
 	REGMAP_ENDIAN_BIG,
 	REGMAP_ENDIAN_LITTLE,
 	REGMAP_ENDIAN_NATIVE,
+	/* Mixed endian: be words in le order (like on PDP-11), 2-1-4-3 */
+	REGMAP_ENDIAN_BIG_LITTLE,
+	/* Mixed endian: le words in be order, 3-4-1-2 */
+	REGMAP_ENDIAN_LITTLE_BIG,
 };
 
 /**
