diff -Naur a/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi b/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi
--- a/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi	2023-09-28 16:05:05.556079967 +0200
@@ -33,6 +33,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			rbpcv2_imx219_a@10 {
+				status = "disabled";
 				reset-gpios = <&gpio CAM1_PWDN GPIO_ACTIVE_HIGH>;
 			};
 		};
@@ -41,6 +42,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			rbpcv2_imx219_e@10 {
+				status = "disabled";
 				reset-gpios = <&gpio CAM2_PWDN GPIO_ACTIVE_HIGH>;
 			};
 		};
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi b/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi
--- a/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi	2023-09-28 10:53:25.599019211 +0200
@@ -23,6 +23,7 @@
     host1x {
 		i2c@546c0000 {
 			rbpcv2_imx219_a@10 {
+				status = "disabled";
 				reset-gpios = <&gpio CAM1_PWDN GPIO_ACTIVE_HIGH>;
 			};
 		};
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx219-dual.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx219-dual.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx219-dual.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx219-dual.dts	2023-09-28 16:31:22.048657973 +0200
@@ -65,6 +65,13 @@
 		};
 	};
 	fragment@6 {
+		target = <&dione_ir_cam0>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
+
+	fragment@7 {
 		target = <&rbpcv2_imx219_vi_in0>;
 		__overlay__ {
 			status = "okay";
@@ -73,7 +80,7 @@
 			remote-endpoint = <&rbpcv2_imx219_csi_out0>;
 		};
 	};
-	fragment@7 {
+	fragment@8 {
 		target = <&rbpcv2_imx219_vi_in1>;
 		__overlay__ {
 			status = "okay";
@@ -82,7 +89,7 @@
 			remote-endpoint = <&rbpcv2_imx219_csi_out1>;
 		};
 	};
-	fragment@8 {
+	fragment@9 {
 		target = <&rbpcv2_imx219_csi_in0>;
 		__overlay__ {
 			status = "okay";
@@ -91,13 +98,13 @@
 			remote-endpoint = <&rbpcv2_imx219_dual_out0>;
 		};
 	};
-	fragment@9 {
+	fragment@10 {
 		target = <&rbpcv2_imx219_csi_out0>;
 		__overlay__ {
 			status = "okay";
 		};
 	};
-	fragment@10 {
+	fragment@11 {
 		target = <&rbpcv2_imx219_csi_in1>;
 		__overlay__ {
 			status = "okay";
@@ -112,4 +119,10 @@
 			status = "okay";
 		};
 	};
+	fragment@13 {
+		target = <&dione_ir_cam1>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
 };
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-dual.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-dual.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-dual.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-dual.dts	2023-09-28 16:04:04.418831814 +0200
@@ -65,7 +65,7 @@
 		};
 	};
 	fragment@6 {
-		target = <&imx219_cam0>;
+		target = <&dione_ir_cam0>;
 		__overlay__ {
 			status = "disabled";
 		};
@@ -120,7 +120,7 @@
 		};
 	};
 	fragment@13 {
-		target = <&imx219_cam1>;
+		target = <&dione_ir_cam1>;
 		__overlay__ {
 			status = "disabled";
 		};
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-imx219.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-imx219.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-imx219.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-all-p3449-0000-camera-imx477-imx219.dts	2023-09-28 16:04:30.099357073 +0200
@@ -65,7 +65,7 @@
 		};
 	};
 	fragment@6 {
-		target = <&imx219_cam0>;
+		target = <&dione_ir_cam0>;
 		__overlay__ {
 			status = "disabled";
 		};
@@ -118,4 +118,10 @@
 			status = "okay";
 		};
 	};
+	fragment@13 {
+		target = <&dione_ir_cam1>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
 };
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx219.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx219.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx219.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx219.dts	2023-09-28 11:14:23.128644516 +0200
@@ -119,4 +119,12 @@
 			isp_bw_margin_pct = <25>;
 		};
 	};
+	/* Disable default sensor */
+	fragment@13 {
+		target = <&dione_ir_single_cam0>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
+
 };
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx477.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx477.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx477.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-common-imx477.dts	2023-09-28 10:56:32.832451456 +0200
@@ -119,7 +119,7 @@
 	};
         /* Disable default sensor */
 	fragment@13 {
-		target = <&imx219_single_cam0>;
+		target = <&dione_ir_single_cam0>;
 		__overlay__ {
 			status = "disabled";
 		};
