From 6071546d1c8a56f2c5ee74eb3d51bb378564edc4 Mon Sep 17 00:00:00 2001
From: Botond Kardos <bkardos70@gmail.com>
Date: Mon, 16 May 2022 13:24:57 +0200
Subject: [PATCH] add "restart_mipi" to sysfs interface

---
 nvidia/drivers/media/i2c/dione_ir.c | 44 +++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/nvidia/drivers/media/i2c/dione_ir.c b/nvidia/drivers/media/i2c/dione_ir.c
index f9929714dc6b..215e4e444b65 100644
--- a/nvidia/drivers/media/i2c/dione_ir.c
+++ b/nvidia/drivers/media/i2c/dione_ir.c
@@ -232,6 +232,8 @@ const struct tc358746_param dioneir_mode_params[] = {
 	dione1280_params
 };
 
+static struct dione_struct *my_dione_struct = NULL;
+
 static int dione1280_i2c_read(struct i2c_client *client, u32 addr, u8 *buf, u16 len);
 static int dione1280_i2c_write32(struct i2c_client *client, u32 addr, u32 val);
 
@@ -1450,12 +1452,44 @@ static struct tegracam_device *dioneir_probe_one( struct dione_struct *priv,
 	return tc_dev;
 }
 
+
+/**
+ * sysfs interface function handling ".../restart_mipi"
+ */
+static ssize_t dioneir_sysfs_restart_mipi(struct kobject *kobj, struct kobj_attribute *attr, const char *buf, size_t count)
+{
+	struct camera_common_data *s_data;
+	int cmd;
+
+	if ( (my_dione_struct != NULL) && (sscanf(buf, "%x", &cmd) == 1) && (cmd == 1)) {
+		s_data = my_dione_struct->s_data;
+		dioneir_stop_streaming( my_dione_struct->tc_dev );
+		msleep(1000);
+		if (s_data->mode == DIONE_IR_MODE_640x480_60FPS)
+			dione640_set_mode( my_dione_struct->tc_dev );
+		else
+			dione1280_set_mode( my_dione_struct->tc_dev );
+
+		dioneir_start_streaming( my_dione_struct->tc_dev );
+	}
+
+	return count;
+}
+
+static struct kobj_attribute dioneir_sysfs_attr_restart_mipi = {
+       .attr = { .name = "restart_mipi", .mode = VERIFY_OCTAL_PERMISSIONS(0220) },
+       .show = NULL,
+       .store = dioneir_sysfs_restart_mipi,
+};
+
+
 static int dioneir_probe(struct i2c_client *client,
 	const struct i2c_device_id *id)
 {
 	struct device *dev = &client->dev;
 	struct tegracam_device *tc_dev;
 	struct dione_struct *priv;
+	char *path;
 	int err;
 
 	dev_dbg(dev, "probing v4l2 sensor at addr 0x%0x\n", client->addr);
@@ -1491,6 +1525,15 @@ static int dioneir_probe(struct i2c_client *client,
 
 		dev_info(dev, "detected Dione IR sensor\n");
 
+		my_dione_struct = priv;
+		err = -EINVAL;
+		path = kobject_get_path(&dev->kobj, GFP_KERNEL);
+		if ( path != NULL )
+			err = sysfs_create_file(&dev->kobj, &dioneir_sysfs_attr_restart_mipi.attr);
+
+		if ( !err )
+			dev_info(dev, "sysfs path: /sys%s/%s\n", path, dioneir_sysfs_attr_restart_mipi.attr.name );
+
 		return 0;
 	}
 
@@ -1504,6 +1547,7 @@ static int dioneir_remove(struct i2c_client *client)
 
 	tegracam_v4l2subdev_unregister(priv->tc_dev);
 	tegracam_device_unregister(priv->tc_dev);
+	sysfs_remove_file(&client->dev.kobj, &dioneir_sysfs_attr_restart_mipi.attr);
 
 	return 0;
 }
-- 
2.25.1

