diff -Naur a/nvidia/platform/t210/batuu/kernel-dts/tegra210-p3448-0003-p3542-0000.dts b/nvidia/platform/t210/batuu/kernel-dts/tegra210-p3448-0003-p3542-0000.dts
--- a/nvidia/platform/t210/batuu/kernel-dts/tegra210-p3448-0003-p3542-0000.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/batuu/kernel-dts/tegra210-p3448-0003-p3542-0000.dts	2023-09-23 15:52:55.667041679 +0200
@@ -21,6 +21,7 @@
 #include "../../porg/kernel-dts/tegra210-porg-p3448-common.dtsi"
 #include "../../porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv3-imx477.dtsi"
 #include "../../porg/kernel-dts/porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi"
+#include "../../porg/kernel-dts/porg-platforms/tegra210-porg-camera-xenics-dione-ir.dtsi"
 #include "batuu-platforms/tegra210-batuu-pinmux-p3448-0003.dtsi"
 #include "batuu-platforms/tegra210-batuu-gpio-p3448-0003.dtsi"
 #include "batuu-platforms/tegra210-batuu-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/Makefile b/nvidia/platform/t210/porg/kernel-dts/Makefile
--- a/nvidia/platform/t210/porg/kernel-dts/Makefile	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/Makefile	2023-09-23 21:53:00.901392949 +0200
@@ -31,9 +31,13 @@
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-0000-p3449-0000-a02-m2ke.dtbo
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-common-imx477.dtbo
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-common-imx219.dtbo
+dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-common-dione-ir.dtbo
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-imx219-dual.dtbo
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-imx477-dual.dtbo
 dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-imx477-imx219.dtbo
+dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-dione-ir-dual.dtbo
+dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-dione-ir-imx219.dtbo
+dtbo-$(CONFIG_ARCH_TEGRA_210_SOC) += tegra210-p3448-all-p3449-0000-camera-dione-ir-imx477.dtbo
 
 ifneq ($(dtb-y),)
 dtb-y := $(addprefix $(makefile-path)/,$(dtb-y))
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-as-p3448-0003.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-as-p3448-0003.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-as-p3448-0003.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-as-p3448-0003.dts	2023-09-23 14:45:57.188999025 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-dual-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dual-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0000-b00.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0000-b00.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a00.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a00.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a00.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a00.dts	2023-09-23 14:45:37.076634681 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0000-a00.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0000-a00.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a01.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a01.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a01.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a01.dts	2023-09-23 14:44:02.914922958 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0000-a01.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0000-a01.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a02.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a02.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a02.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-a02.dts	2023-09-23 14:45:17.044271360 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0000-a02.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0000-a02.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-b00.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-b00.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-b00.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0000-p3449-0000-b00.dts	2023-09-23 14:44:56.259893931 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-dual-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dual-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0000-b00.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0000-b00.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-a02.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-a02.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-a02.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-a02.dts	2023-09-23 14:43:38.018468554 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0002-a02.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0002-a02.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
diff -Naur a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-b00.dts b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-b00.dts
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-b00.dts	2021-07-26 21:08:25.000000000 +0200
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-p3448-0002-p3449-0000-b00.dts	2023-09-23 14:44:26.419351200 +0200
@@ -21,6 +21,7 @@
 #include "tegra210-porg-p3448-common.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv3-dual-imx477.dtsi"
 #include "porg-platforms/tegra210-porg-camera-rbpcv2-dual-imx219.dtsi"
+#include "porg-platforms/tegra210-porg-camera-xenics-dual-dione-ir.dtsi"
 #include "porg-platforms/tegra210-porg-pinmux-p3448-0002-b00.dtsi"
 #include "porg-platforms/tegra210-porg-gpio-p3448-0002-b00.dtsi"
 #include "porg-platforms/tegra210-porg-p3448-emc-a00.dtsi"
